src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
> Isn't it a programmer's error to end up in this situation?

Yes and it doesn't make your life any better when deployed app silently hangs in production. You may need to re-read https://github.com/sociomantic/backend/blob/master/doc/coding/assert-vs-enforce.rst :P